addons/crm/models/crm_lead.py

```diff
@@ -240,6 +240,30 @@ def _compute_partner_id_values(self):
         for lead in self:
             lead.update(lead._preare_values_from_partner(lead.partner_id))
 
+    @api.depends('partner_id.email')
+    def _compute_email_from(self):
+        for lead in self:
+            if lead.partner_id and lead.partner_id.email != lead.email_from:
+                lead.email_from = lead.partner_id.email
+
+    def _inverse_email_from(self):
+        for lead in self:
+            if lead.partner_id and lead.email_from != lead.partner_id.email:
+                lead.partner_id.email = lead.email_from
+
+    @api.depends('partner_id.phone')
+    def _compute_phone(self):
+        with mute_logger('odoo.addons.phone_validation.tools.phone_validation'):
+            for lead in self:
+                if lead.partner_id and self.phone_format(lead.partner_id.phone) != self.phone_format(lead.phone):
+                    lead.phone = lead.partner_id.phone
```

Comment:
Lead inherits form mail.thread.phone, and partner from phone.validation.mixin . I bet this code can be improved a lot !